source/utils/types.ts

```diff
@@ -9,10 +9,35 @@ export type NextFunction = (error?: Error | string) => void;
 
 export type IterateFunction = (options: Options) => void;
 
+export interface Response extends IncomingMessage {
+	body: string | Buffer;
+	statusCode: number;
+}
+
+export interface Timings {
+	start: number;
+	socket: number | null;
+	lookup: number | null;
+	connect: number | null;
+	upload: number | null;
+	response: number | null;
+	end: number | null;
+	error: number | null;
+	phases: {
+		wait: number | null;
+		dns: number | null;
+		tcp: number | null;
+		request: number | null;
+		firstByte: number | null;
+		download: number | null;
+		total: number | null;
+	};
+}
+
 export interface Instance {
-	methods?: Method[];
-	options?: Options;
-	handler?: (options: Options, callback: NextFunction) => void;
+	methods: Method[];
+	options: Partial<Options> & { hooks: Partial<Hooks> };
```

Comment:
I don't know if it is correct instances only contains parts of options or it contains _all_ options.

Reply:
Parts. Also isn't `Partial<Options>` enough? It has defined `hooks: Partial<Hooks>` already...